text/0001-variables.md

```diff
@@ -0,0 +1,198 @@
+- Feature Name: Variables in Blight
+- Start Date: 2023-11-03
+- RFC PR: [blight-js/rfcs#0001](https://github.com/blight-js/rfcs/pull/2)
+- Blight Issue: https://github.com/blight-js/blight/issues/2
+
+# Summary
+
+[summary]: #summary
+
+We at Blight don't believe that functional programming languages make sense.
+Blight is not a functional programming language and thus relies on some form of
+state storage. We solve this using variables.
+
+# Motivation
+
+[motivation]: #motivation
+
+To make interesting programs, one needs to be able to store state. Adding state
+storage allows for traditional declarative programming. Being able to store
+state will allow for future work around variable semantics.
+
+# Guide-level explanation
+
+[guide-level-explanation]: #guide-level-explanation
+
+This proposal adds variables to the language. Variables are a way to store state between operations.
+
+This RFC only defines immutable variables. All immutable values are deeply immutable. Syntax for mutable variables may be defined in a future RFC.
+
+
+Let's look at an example of variables in the context of Blight:
+
+```blight
+let <variable_name>: <type> = <value>;
+```
+
+i.e. with an example type of "u8" and an example value of "10"
+
+```blight
+let x: u8 = 10;
+```
+
+The type and value stored inside of variables will be defined in a future
+proposal.
+
+The variable name can be any valid unicode except for the following conditions:
+
+- Variables may not start with numerical digits
+- Variables may not have any whitespace characters
+- Variables may not contain with any of the following characters as they are
+  reserved for future language use: "'<>(){}[]\|/?,.=+-~`!@#$%^&*
+
+```blight
+// Valid
+let x: <type> = <value>;
```

Comment:
Is there a reason let is used and not const, as variables are by default immutable? Or is this part of the philosophy, so the immutability doesn't have to be implied?

Reply:
I think it's part of the philosophy. This was... hotly debated. We're going with `let` for now and we'll revisit that decision later.